examples/shopping/carts.rst

```diff
@@ -0,0 +1,107 @@
+.. _shopping-carts:
+
+Shopping Carts
+==============
+
+At this point, we have Captured Collections setup for users and products, and we're ready to
+start building the shopping carts. We'll start by defining a Captured Collection for cart
+updates, which will hold a record of each time a user adds or modifies a product in their cart.
+These cart updates will then be joined with the product information so that we'll have the price
+of each item. Then we'll create a "carts" Collection that rolls up all the joined updates into a
+single document that includes all the items in a users cart, along with their prices.
+
+Summing Item Quantities
+-----------------------
+
+Here's the Collection and schema for the cart updates:
+
+.. literalinclude:: cart.flow.yaml
+    :language: yaml
+    :lines: 5-7
+
+.. literalinclude:: schema.yaml
+    :language: yaml
+    :lines: 13-22
+    :emphasize-lines: 20
+
+Each updateCart document represents a request to add or subtract a quantity of a particular
+product in a user's cart. If we get multiple updates for the same user and product, the
+quantities will be summed. This is because of the ``reduce`` annotation in the above schema.
+
+
+Joining Cart Updates and Products
+---------------------------------
+
+We'll define a new :ref:`derived collection <concepts-derivations>` that performs a streaming
+inner join of ``updateCart`` to ``product`` documents.
+
+.. literalinclude:: cart.flow.yaml
+    :language: yaml
+    :lines: 13-48
+
+There's two main concepts being used here. The first is the :ref:`register <concepts-registers>`. We'll have a unique register value for each product id, since that's the value we're joining on. For each product id, the register value can either be a product document, or null, and the ``initial`` value is always ``null``.
+
+Now let's look at the two :ref:`transforms <concepts-derivations>`, starting with ``products``.
+This will read documents from the products Collection, and update the register for each one. Note
+that the default shuffle key is implicitly the key of the source collection, in this case the
+``/id`` field of a ``product``. The return value of ``[source]`` might seem a bit magical, but
```

Comment:
nit: hah probably shouldn't be describing something we're doing as "magical" in our own documents.

Prospective users might accuse us of too much "magic", but we don't need to plant that seed for them.